sdk/tests/core/test_az_mqtt5_connection.c

```diff
@@ -434,28 +668,917 @@ static void test_az_mqtt5_connection_open_idle_success(void** state)
   will_return(__wrap_az_platform_clock_msec, 0);
   will_return(__wrap_az_platform_clock_msec, 0);
-  will_return(__wrap_az_platform_clock_msec, 0);
-  assert_int_equal(az_mqtt5_connection_open(&test_connection), AZ_OK);
+  will_return(__wrap_az_mqtt5_outbound_disconnect, AZ_OK);
+  will_return(__wrap_az_platform_timer_create, AZ_OK);
+  assert_int_equal(az_mqtt5_connection_close(&test_connection), AZ_OK);
+  _az_mqtt5_connection_disconnecting_idle(&mock_mqtt);
+
+  assert_int_equal(ref_disconn_req, 1);
+  assert_int_equal(ref_disconn_rsp, 2);
+  assert_int_equal(ref_conn_error, 0);
+}
+
+// Testing that when connecting no publish requests are allowed.
+static void test_az_mqtt5_connection_connecting_pub_request_success(void** state)
+{
+  (void)state;
+  _reset_refs();
+  az_mqtt5_connection test_connection;
+  az_mqtt5 mock_mqtt;
+  az_mqtt5_options mock_mqtt_options = { 0 };
+  _az_event_client mock_client_1;
+  _az_event_client mock_client_2;
+  _az_hfsm mock_client_hfsm_1;
+  _az_hfsm mock_client_hfsm_2;
+
+  _az_mqtt5_connection_setup(
+      &test_connection,
+      &test_connection_options,
+      &mock_mqtt,
+      &mock_mqtt_options,
+      &mock_client_1,
+      &mock_client_2,
+      &mock_client_hfsm_1,
+      &mock_client_hfsm_2);
+
+  _az_mqtt5_connection_idle_connecting(&test_connection);
+
+  assert_int_equal(
+      _az_event_pipeline_post_outbound_event(
+          &test_connection._internal.event_pipeline,
+          (az_event){ .type = AZ_MQTT5_EVENT_PUB_REQ,
+                      .data = &(az_mqtt5_pub_data){ .topic = AZ_SPAN_FROM_STR(TEST_TOPIC),
+                                                    .qos = AZ_MQTT5_QOS_AT_MOST_ONCE,
+                                                    .payload = AZ_SPAN_FROM_STR(TEST_PAYLOAD) } }),
+      AZ_ERROR_HFSM_INVALID_STATE);
+}
+
+// Testing that when connecting, incoming publishes are received and processed.
+static void test_az_mqtt5_connection_connecting_pub_recv_success(void** state)
+{
+  (void)state;
+  _reset_refs();
+  az_mqtt5_connection test_connection;
+  az_mqtt5 mock_mqtt;
+  az_mqtt5_options mock_mqtt_options = { 0 };
+  _az_event_client mock_client_1;
+  _az_event_client mock_client_2;
+  _az_hfsm mock_client_hfsm_1;
+  _az_hfsm mock_client_hfsm_2;
+
+  _az_mqtt5_connection_setup(
+      &test_connection,
+      &test_connection_options,
+      &mock_mqtt,
+      &mock_mqtt_options,
+      &mock_client_1,
+      &mock_client_2,
+      &mock_client_hfsm_1,
+      &mock_client_hfsm_2);
+
+  _az_mqtt5_connection_idle_connecting(&test_connection);
+
+  assert_int_equal(
+      az_mqtt5_inbound_recv(
+          &mock_mqtt,
+          &(az_mqtt5_recv_data){ .topic = AZ_SPAN_FROM_STR(TEST_TOPIC),
+                                 .qos = AZ_MQTT5_QOS_AT_MOST_ONCE,
+                                 .payload = AZ_SPAN_FROM_STR(TEST_PAYLOAD),
+                                 .id = 1 }),
+      AZ_OK);
+
+  // TODO_L: Once event filtering is implemented, add more details.
```

Comment:
can you do the same checks here as in `test_az_mqtt5_connection_connected_pub_rcv_success`?

Reply:
There currently is no event filtering in the root state so I can't check that the pub rcv was sent to the application but I will add the other two checks